do-not-commit/pickle.py

```diff
@@ -0,0 +1,65 @@
+# Import dependencies
+import _pickle
+import os
+
+# Attacker prepares exploit that application will insecurely deserialize
+class Exploit(object):
+    def __reduce__(self):
+        return (os.system, ("whoami",))
+
+
+# Attacker serializes the exploit
+def serialize_exploit():
+    # ruleid: avoid-pickle
+    shellcode = _pickle.dumps(Exploit())
```

Comment:
Avoid using `pickle`, which is known to lead to code execution vulnerabilities.
When unpickling, the serialized data could be manipulated to run arbitrary code.
Instead, consider serializing the relevant data as JSON or a similar text-based
serialization format. ([serialization.pickle.avoid-pickle](https://semgrep.dev/r/serialization.pickle.avoid-pickle))
#
<div>
<a href="https://semgrep.dev">
<img align="right" src="https://r2c.dev/semgrep_icon_text.svg" height=28px alt="" />
</a>

<sub> [How do I suppress this?](https://semgrep.dev/docs/running-rules/#managing-findings) **·** [Report a bug](https://github.com/returntocorp/semgrep-rules/issues/new?assignees=&labels=&template=rule_bug_report.md&title=serialization.pickle.avoid-pickle) **·** [Manage rule](https://semgrep.dev/manage/projects)</sub>
</div>